src/Math-Tests-Complex/PMComplexTest.class.st

```diff
@@ -783,6 +729,72 @@ PMComplexTest >> testTanh [
 	self assert: (c2 imaginary closeTo: c tanh imaginary).
 ]
 
+{ #category : #'testing - complex conjugation' }
+PMComplexTest >> testThatComplexConjugateHasEqualRealPartAndAnImaginaryPartWithEqualMagnitudeButOppositeSign [
+	| z |
+	z := 3 + 2 i.
+	
+	self assert: z complexConjugate equals: 3 - 2 i
+]
+
+{ #category : #'testing - complex conjugation' }
+PMComplexTest >> testThatComplexConjugateOfAPureRealNumberIsItself [
+	| z |
+	z := 5 + 0 i.
+	
+	self assert: z complexConjugate equals: 5.
```

Comment:
This test is not good.

First, the same comment about test name as above.
I would use a shorter name: `testComplexConjugateOfRealNumber`
The fact that it's itself is a definition of complex conjugate and should be in the method comment and not the test name

Second, about the temporary variable, same comment as above - I would not use it because it is unnecessary and make the code harder to read (of course, this is subjective).

Third (!), this test is misleading.
`(5 + 0i) complexConjugate` does not return `5` (a real number)
It returns `(5 + 0i)` (a complex number)

The reason why this test passes is because `(5 + 0i) = 5`

--------------
I would do the following:

1. Remove this test, because for complex numbers, testing that `self assert: (5 - 0i) complexConjugate equals (5 + 0i)` is the same test as `self assert: (3 + 2i) complexConjugate equals (3 - 2i)`. And we already have that test as `testComplexConjugate`
2. Add `complexConjugate` to `Number` class (see my first comment to this review)
3. Write the following test for `Number`:

```st
NumberTest >> testComplexConjugate

	self assert: 5 complexConjugate equals: 5
```